### DISCUSSION TITLE
Permissions

### DISCUSSION THREAD

**Discussion by ccsv** [reset-button/ariadne_django#57] [General]
Just looking at the permissions to see what we can and cannot do

Is there a way to restrict the fields in query results based on users permission?

There are two decorators in the auth folder `@login_required()` and `@permission_required` but I have not used them

**Comment by maxmorlocke** [comment]
I believe graphql directives should provide the solution you're looking for in terms of restricting on query results.  The decorators are generally targeted at the endpoint.

https://ariadnegraphql.org/docs/schema-directives
